security/signup-controller.php

```diff
@@ -0,0 +1,38 @@
+<?php
+include "db-connection.php";
+
+$conn = openCon();
+
+$firstname = isset($_POST['firstname']) ? $_POST['firstname'] : false;
+$password = isset($_POST['password']) ? $_POST['password'] : false;
+$role = isset($_POST['role']) ? $_POST['role'] : false;
+if ( strcmp($role, "admin")==0 ) die("You are not authorized to do this.");
+
+if ($firstname && $password && $role)
+{
+  $email = $conn->prepare("SELECT COUNT(*) FROM person WHERE email = :emailPlaceholder");
```

Comment:
I like that you used prepare statement. You an extract sql query in separate variable to make the code more readable.